src/transforms/aws_ec2_metadata.rs

```diff
@@ -516,6 +535,24 @@ impl MetadataClient {
                 }
             }
 
+            for tag in self.tags.clone() {
```

Comment:
I don't think this needs a `clone` here. You should be able to use a reference:
```suggestion
            for tag in &self.tags {
```

Reply:
We can't borrow `self` here because `self.get_metadata` borrows `&mut self` inside the loop.